text/0000-ledger-api.md

```diff
@@ -0,0 +1,380 @@
+---
+layout: default
+title: Ledger_API
+parent: RFCs
+---
+
+- Feature Name: Ledger_API
+- Start Date: 2019-12-03
+- RFC PR: (leave this empty)
+- Fabric Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This feature is a continuation of the updates made in recent releases to the Fabric Programming Model. This RFC's aim is to clarify the abstractions used to work with ledger data in a contract. It's scope is restricted to what is today referred to as the 'ChaincodeStub API' specifically the APIs around states.
+
+This is a cross-language RFC for Golang, JavaScript, TypeScript and Java.
+
+# Motivation
+[motivation]: #motivation
+
+The updated Programming Model has introduced higher level abstractions to both the contract and the client SDKs. Many users have already written high-level utility or support code within their own applications. Abstractions help us think about and solve problems, improving usability, speed of development and performance.
+
+Examples, such as the Commercial Paper tutorial, also take this approach. It simplifies the concepts and makes working with business-level data simpler. (c.f. https://jira.hyperledger.org/browse/FABCN-42)
+
+This RFC (which is anticipated as the first in a series) will formalize the existing 'stub-api' into a conceptual architecture. A key aspect in all the programming model work is that the conceptual model is consistent between languages. The polyglot developer will therefore have minimal friction moving between languages.  Note that this a *conceptual* model; each implementation will use idiomatic language code, and where appropriate, specifics of the language to give the best developer experience.
+
+The stub API is broadly consistent between languages. There are, however, variations and minor inconsistencies that cause confusion when looking across languages. Whilst not major, these differences are sufficient to cause a developer uncertainty in how an API should be used. Introducing a ledger formalizes and integrates existing concepts. 
```

Comment:
Curious to know about these inconsistencies. An example would be better.

Reply:
I've also added some links to JIRAs later in the RFC.

A lot of these are minor in the spelling of the APIs varies between languages. 
The return type of the iterators is some APIs in node.js wrapped in other objects. 
APIs in Java have been missed out for example.